addons/image/bmp.c

```diff
@@ -1180,4 +1180,28 @@ bool _al_save_bmp(const char *filename, ALLEGRO_BITMAP *bmp)
    return retsave && retclose;
 }
 
+
+bool _al_identify_bmp(ALLEGRO_FILE *f)
+{
+   uint16_t x;
+   x = al_fread16le(f);
+   if (x != 'B' + 'M' * 256)
```

Comment:
I feel like doing a memcmp here just like with the other identifiers might be easier to look at.